src/axolotl/utils/callbacks/perplexity.py

```diff
@@ -17,49 +19,51 @@ class Perplexity:
 
     def __init__(
         self,
-        model: PreTrainedModel,
         tokenizer: PreTrainedTokenizer,
         max_seq_len: int,
         stride: int = 512,
     ) -> None:
         self.max_seq_len = max_seq_len
         self.stride = stride
-        self.model = model
         self.tokenizer = tokenizer
-        self.device = model.device
         self.name = "perplexity"
 
     def _feature_names(self) -> List[str]:
         return ["references"]
 
     def compute(
         self,
+        model: PreTrainedModel,
         references: Optional[List[str]] = None,
     ) -> Dict[str, float]:
         """
         Compute perplexity in a fixed length sliding window across the sequence.
         """
         assert references is not None, "Missing parameter: references"
 
+        model.eval()
```

Comment:
if we set model.eval() here, do we need to switch it back to model.train at the end?

Reply:
Should be no. This plugin  is only run `on_evaluate` from the `CausalLMBenchEvalCallback`. In fact, we already have one at the start too, so this line may not be needed.

If you're worried, we can check its state and set it back to it after. I would actually be cautious as that would make any runs after while `on_evaluate` to have a potential model in training mode.